Test.Microsoft.NET.Build.Containers/ImageBuilderTests.cs

```diff
@@ -0,0 +1,294 @@
+                    "config": {
+                      "Hostname": "",
+                      "Domainname": "",
+                      "User": "",
+                      "AttachStdin": false,
+                      "AttachStdout": false,
+                      "AttachStderr": false,
+                      "Tty": false,
+                      "OpenStdin": false,
+                      "StdinOnce": false,
+                      "Env": [
+                        "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+                        "ASPNETCORE_URLS=http://+:80",
+                        "DOTNET_RUNNING_IN_CONTAINER=true",
+                        "DOTNET_VERSION=7.0.2",
+                        "ASPNET_VERSION=7.0.2"
+                      ],
+                      "Cmd": ["bash"],
+                      "Image": "sha256:d772d27ebeec80393349a4770dc37f977be2c776a01c88b624d43f93fa369d69",
+                      "Volumes": null,
+                      "WorkingDir": "",
+                      "Entrypoint": null,
+                      "OnBuild": null,
+                      "Labels": null,
+                      "ExposedPorts":
+                      {
+                        "6100/tcp": {}
+                      }
+                    },
+                    "created": "2023-02-04T08:14:52.000901321Z",
+                    "os": "linux",
+                    "rootfs": {
+                      "type": "layers",
+                      "diff_ids": [
+                        "sha256:bd2fe8b74db65d82ea10db97368d35b92998d4ea0e7e7dc819481fe4a68f64cf",
+                        "sha256:94100d1041b650c6f7d7848c550cd98c25d0bdc193d30692e5ea5474d7b3b085",
+                        "sha256:53c2a75a33c8f971b4b5036d34764373e134f91ee01d8053b4c3573c42e1cf5d",
+                        "sha256:49a61320e585180286535a2545be5722b09e40ad44c7c190b20ec96c9e42e4a3",
+                        "sha256:8a379cce2ac272aa71aa029a7bbba85c852ba81711d9f90afaefd3bf5036dc48"
+                      ]
+                    }
+                }
+                """;
+
+            JsonNode? node = JsonNode.Parse(simpleImageConfig);
+            Assert.IsNotNull(node);
+
+            ImageConfig baseConfig = new ImageConfig(node);
+
+            baseConfig.ExposePort(6000, PortType.tcp);
+            baseConfig.ExposePort(6010, PortType.udp);
+            baseConfig.ExposePort(6100, PortType.udp);
+
+            string readyImage = baseConfig.BuildConfig();
+
+            JsonNode? result = JsonNode.Parse(readyImage);
+
+            var resultPorts = result?["config"]?["ExposedPorts"] as JsonObject;
+            Assert.IsNotNull(resultPorts);
+
+            // Assert.AreEqual(3, resultPorts.Count);   --> 4 6100/tcp also exists
+            Assert.IsNotNull(resultPorts["6000/tcp"] as JsonObject);
+            Assert.IsNotNull(resultPorts["6010/udp"] as JsonObject);
+            Assert.IsNotNull(resultPorts["6100/udp"] as JsonObject);
```

Comment:
@baronfel is it correct behavior? should 6100 be replaced in this case?

Reply:
the `key` for the port mapping is two parts: required port number and optional port type (`tcp` or `udp`). if no port type is specified than `tcp` is assumed by the engine (not necessarily by us at this layer). In addition, multiple protocols _can_ bind to the same port, so adding a new entry with the same port but different protocol is allowed.

So adding a `6100/udp` in this case would represent 4 total port bindings, 600/tcp, 6010/udp, and 6100 on both tcp and udp.

Reply:
The current implementation assumes that if no port type is specified, the port type is `tcp`.
Assuming replacing `6200` with `6200/tcp` as in this test is ok.